Minified Dep Bundles are Broken

When building the CanJS site if I use `"deps-bundle": "steal-tools bundle --deps"` in my scripts and try to load a demo with that deps-bundle like:
```
<script id="demo-source" main="@empty" src="../../node_modules/steal/steal.js" deps-bundle>
```
It will break on load with the error:
```
steal.js:7390 Error: Error loading "dev-bundle" at http://localhost:8080/dev-bundle.js
TypeError: define is not a function
	    at Object.eval (http://localhost:8080/dev-bundle.js:4:125077)
	    at eval (http://localhost:8080/dev-bundle.js:5:4)
	    at eval (http://localhost:8080/dev-bundle.js:5:98)
	Evaluating http://localhost:8080/dev-bundle.js
```

If I add the `--no-minify` option `"deps-bundle": "steal-tools bundle --deps --no-minify"` it works fine but the deps-bundle is not minified.

--- a/.gitignore
+++ b/.gitignore
@@ -40,3 +40,4 @@ test/transform_export/out.js
 test/exports_basics/out.js
 test/circular/export/
 test/serviceworker/service-worker.js
+!test/dev_bundles_minify/node_modules

--- a/lib/bundle/add_npm_packages_node.js
+++ b/lib/bundle/add_npm_packages_node.js
@@ -1,3 +1,4 @@
+var prettier = require("prettier");
 
 /**
  * Adds node to bundle to preload npm packages for dev bundles
@@ -6,11 +7,8 @@
  * @param {{}} npmContext The context containing the npm packages loaded
  */
 module.exports = function(bundle, npmContext) {
-	var unshift = [].unshift;
-
-	unshift.apply(bundle.nodes, [
-		makeAddNpmPackagesNode(npmContext)
-	]);
+	var push = [].push;
+	push.apply(bundle.nodes, [makeAddNpmPackagesNode(npmContext)]);
 };
 
 function makeAddNpmPackagesNode(npmContext) {
@@ -23,11 +21,12 @@ function makeAddNpmPackagesNode(npmContext) {
 			metadata: {
 				format: "global"
 			},
-			source: [
-				"if (steal && typeof steal.addNpmPackages === 'function') {",
-				"steal.addNpmPackages(" + JSON.stringify(packages) + ");",
-				"}"
-			].join(" ")
+			source: prettier.format(
+				`if (steal && typeof steal.addNpmPackages === "function") {
+					steal.addNpmPackages(${JSON.stringify(packages)});
+				}`,
+				{ tabWidth: 4 }
+			)
 		}
 	};
 }

--- a/test/dev_bundle_build_test.js
+++ b/test/dev_bundle_build_test.js
@@ -344,4 +344,38 @@ describe("dev bundle build", function() {
 				return rmdir(devBundlePath);
 			});
 	});
+
+	it("minified dev bundles work", function(done) {
+		var dir = path.join(__dirname, "dev_bundles_minify");
+		var devBundlePath = path.join(dir, "dev-bundle.js");
+
+		var config = {
+			config: path.join(dir, "package.json!npm")
+		};
+
+		var options = assign({}, baseOptions, {
+			minify: true,
+			filter: "node_modules/**/*" // only bundle npm deps
+		});
+
+		var clean = function(err) {
+			rmdir(devBundlePath).then(function() {
+				done(err);
+			});
+		};
+		devBundleBuild(config, options)
+			.then(function() {
+				var exists = fs.existsSync(devBundlePath);
+				assert(exists, "dev bundle should be created");
+			})
+			.then(function() {
+				return open("test/dev_bundles_minify/dev.html");
+			})
+			.then(function(p) {
+				p.close();
+				p.browser.assert.element("h1");
+			})
+			.then(clean)
+			.catch(clean);
+	});
 });

--- a/test/dev_bundles_minify/dev.html
+++ b/test/dev_bundles_minify/dev.html
@@ -1,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+	<meta charset="UTF-8">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<meta http-equiv="X-UA-Compatible" content="ie=edge">
+	<title>Minified development bundles</title>
+</head>
+<body>
+	<script src="../../node_modules/steal/steal-sans-promises.js"
+		base-url="."
+		config-main="package.json!npm"
+		deps-bundle="dev-bundle"
+	></script>
+</body>
+</html>

--- a/test/dev_bundles_minify/main.js
+++ b/test/dev_bundles_minify/main.js
@@ -1,0 +1,6 @@
+var dummy = require("dummy");
+dummy();
+
+var h1 = document.createElement("h1");
+h1.innerHTML = "worked!";
+document.body.appendChild(h1);

--- a/test/dev_bundles_minify/package.json
+++ b/test/dev_bundles_minify/package.json
@@ -1,0 +1,8 @@
+{
+	"name": "app",
+	"main": "main.js",
+	"version": "1.0.0",
+	"dependencies": {
+		"dummy": "1.0.0"
+	}
+}

